From 6526855d7f8e962f118e54eb17bb445bc8338e0c Mon Sep 17 00:00:00 2001
From: Matthias Reichl <hias@horus.com>
Date: Sun, 12 Mar 2017 17:06:53 +0100
Subject: [PATCH 1/2] v4l-utils: create separate keymaps instead of patching
 rc6_mce

All locally added remote keycodes are now in separate files,
one per remote, instead of adding them to rc6_mce.

A new keymap, libreelec_multi, with all keycodes of the previous
rc6_mce map is added in install and used by default when the
remote is configured to use the rc-rc6-mce table.

The default behaviour is unchanged, LE can still be controlled with
MCE, xbox, ... remotes out of the box but it's now possible to configure
the ir receiver to react only on the codes sent by a specific remote.

This solves the issue where a xbox remote always controlled LE even
when a user only wanted to control it with a MCE remote.

To do this simply create a /storage/.config/rc_maps.cfg file with
the following content:

*	rc-rc6-mce	rc6_mce

Signed-off-by: Matthias Reichl <hias@horus.com>
---
 packages/sysutils/v4l-utils/package.mk             |  8 +++
 .../v4l-utils-11-fix_keytables_xbox360-0.1.patch   | 73 ----------------------
 .../v4l-utils-12-add_zbox_ad10_support-0.1.patch   | 52 ---------------
 .../v4l-utils-20-add-samsung-keytable.patch        | 46 ++++++++++++++
 .../v4l-utils-20-add_samsung_keytable.patch        | 46 --------------
 .../v4l-utils-21-add-newer-hp-mce-remote.patch     | 17 +++--
 .../patches/v4l-utils-22-add-XBOX_One-remote.patch | 36 -----------
 .../patches/v4l-utils-22-add-xbox-one-remote.patch | 28 +++++++++
 .../patches/v4l-utils-23-add-cubox-i-remote.patch  | 17 +++--
 .../patches/v4l-utils-24-add-xbox-360-remote.patch | 72 +++++++++++++++++++++
 .../v4l-utils-25-add-zotac-ad10-remote.patch       | 50 +++++++++++++++
 ...-utils-50-use-multi-table-add-fake-tables.patch | 29 +++++++++
 12 files changed, 247 insertions(+), 227 deletions(-)
 delete mode 100644 packages/sysutils/v4l-utils/patches/v4l-utils-11-fix_keytables_xbox360-0.1.patch
 delete mode 100644 packages/sysutils/v4l-utils/patches/v4l-utils-12-add_zbox_ad10_support-0.1.patch
 create mode 100644 packages/sysutils/v4l-utils/patches/v4l-utils-20-add-samsung-keytable.patch
 delete mode 100644 packages/sysutils/v4l-utils/patches/v4l-utils-20-add_samsung_keytable.patch
 delete mode 100644 packages/sysutils/v4l-utils/patches/v4l-utils-22-add-XBOX_One-remote.patch
 create mode 100644 packages/sysutils/v4l-utils/patches/v4l-utils-22-add-xbox-one-remote.patch
 create mode 100644 packages/sysutils/v4l-utils/patches/v4l-utils-24-add-xbox-360-remote.patch
 create mode 100644 packages/sysutils/v4l-utils/patches/v4l-utils-25-add-zotac-ad10-remote.patch
 create mode 100644 packages/sysutils/v4l-utils/patches/v4l-utils-50-use-multi-table-add-fake-tables.patch

diff --git a/packages/sysutils/v4l-utils/package.mk b/packages/sysutils/v4l-utils/package.mk
index 28c1803..e4f5f27 100644
--- a/packages/sysutils/v4l-utils/package.mk
+++ b/packages/sysutils/v4l-utils/package.mk
@@ -45,4 +45,12 @@ post_makeinstall_target() {
 
   mkdir -p $INSTALL/usr/config
     cp -PR $PKG_DIR/config/* $INSTALL/usr/config
+
+  (
+    echo "# table libreelec_multi, type: RC6 NEC"
+    for f in rc6_mce xbox_360 zotac_ad10 hp_mce xbox_one cubox_i ; do
+      echo "# $f"
+      grep -v "^#" $INSTALL/usr/lib/udev/rc_keymaps/$f
+    done
+  ) > $INSTALL/usr/lib/udev/rc_keymaps/libreelec_multi
 }
diff --git a/packages/sysutils/v4l-utils/patches/v4l-utils-11-fix_keytables_xbox360-0.1.patch b/packages/sysutils/v4l-utils/patches/v4l-utils-11-fix_keytables_xbox360-0.1.patch
deleted file mode 100644
index 481d17d..0000000
--- a/packages/sysutils/v4l-utils/patches/v4l-utils-11-fix_keytables_xbox360-0.1.patch
+++ /dev/null
@@ -1,73 +0,0 @@
-diff -rupN v4l-utils-0.8.3/utils/keytable/rc_keymaps/rc6_mce v4l-utils-0.8.3.patch/utils/keytable/rc_keymaps/rc6_mce
---- v4l-utils-0.8.3/utils/keytable/rc_keymaps/rc6_mce	2011-08-19 19:08:11.000000000 +0200
-+++ v4l-utils-0.8.3.patch/utils/keytable/rc_keymaps/rc6_mce	2011-09-22 21:50:44.000000000 +0200
-@@ -60,3 +60,69 @@
- 0x800f046f KEY_MEDIA
- 0x800f0480 KEY_BRIGHTNESSDOWN
- 0x800f0481 KEY_PLAYPAUSE
-+#xbox360 remote
-+0x800f7400 KEY_NUMERIC_0
-+0x800f7401 KEY_NUMERIC_1
-+0x800f7402 KEY_NUMERIC_2
-+0x800f7403 KEY_NUMERIC_3
-+0x800f7404 KEY_NUMERIC_4
-+0x800f7405 KEY_NUMERIC_5
-+0x800f7406 KEY_NUMERIC_6
-+0x800f7407 KEY_NUMERIC_7
-+0x800f7408 KEY_NUMERIC_8
-+0x800f7409 KEY_NUMERIC_9
-+0x800f740a KEY_DELETE
-+0x800f740b KEY_ENTER
-+0x800f740c KEY_SLEEP
-+0x800f740d KEY_MEDIA
-+0x800f7464 KEY_MEDIA
-+0x800f740e KEY_MUTE
-+0x800f740f KEY_INFO
-+0x800f7410 KEY_VOLUMEUP
-+0x800f7411 KEY_VOLUMEDOWN
-+0x800f7412 KEY_CHANNELUP
-+0x800f746C KEY_CHANNELUP
-+0x800f7413 KEY_CHANNELDOWN
-+0x800f746D KEY_CHANNELDOWN
-+0x800f7414 KEY_FASTFORWARD
-+0x800f7415 KEY_REWIND
-+0x800f7416 KEY_PLAY
-+0x800f7417 KEY_RECORD
-+0x800f7418 KEY_PAUSE
-+0x800f7419 KEY_STOP
-+0x800f741a KEY_NEXT
-+0x800f741b KEY_PREVIOUS
-+0x800f741c KEY_NUMERIC_POUND
-+0x800f741d KEY_NUMERIC_STAR
-+0x800f741e KEY_UP
-+0x800f741f KEY_DOWN
-+0x800f7420 KEY_LEFT
-+0x800f7421 KEY_RIGHT
-+0x800f7422 KEY_OK
-+0x800f7423 KEY_EXIT
-+0x800f7424 KEY_DVD
-+0x800f744f KEY_EPG
-+0x800f7427 KEY_ZOOM
-+0x800f7432 KEY_MODE
-+0x800f7433 KEY_PRESENTATION
-+0x800f7428 KEY_EJECTCD
-+0x800f743a KEY_BRIGHTNESSUP
-+0x800f7446 KEY_TV
-+0x800f7447 KEY_AUDIO
-+0x800f7448 KEY_PVR
-+0x800f7449 KEY_CAMERA
-+0x800f744a KEY_VIDEO
-+0x800f744c KEY_LANGUAGE
-+0x800f7451 KEY_TITLE
-+0x800f744e KEY_PRINT
-+0x800f7450 KEY_RADIO
-+0x800f745a KEY_SUBTITLE
-+0x800f7425 KEY_RED
-+0x800f7466 KEY_GREEN
-+0x800f7426 KEY_YELLOW
-+0x800f7468 KEY_BLUE
-+0x800f7465 KEY_POWER2
-+0x800f746e KEY_PLAYPAUSE
-+0x800f746f KEY_PLAYER
-+0x800f7480 KEY_BRIGHTNESSDOWN
-+0x800f7481 KEY_PLAYPAUSE
diff --git a/packages/sysutils/v4l-utils/patches/v4l-utils-12-add_zbox_ad10_support-0.1.patch b/packages/sysutils/v4l-utils/patches/v4l-utils-12-add_zbox_ad10_support-0.1.patch
deleted file mode 100644
index 6bf5d02..0000000
--- a/packages/sysutils/v4l-utils/patches/v4l-utils-12-add_zbox_ad10_support-0.1.patch
+++ /dev/null
@@ -1,52 +0,0 @@
-diff -Naur v4l-utils-0.8.3/utils/keytable/rc_keymaps/rc6_mce v4l-utils-0.8.3.patch/utils/keytable/rc_keymaps/rc6_mce
---- v4l-utils-0.8.3/utils/keytable/rc_keymaps/rc6_mce	2013-09-22 13:54:18.609413831 +0200
-+++ v4l-utils-0.8.3.patch/utils/keytable/rc_keymaps/rc6_mce	2013-09-22 13:56:37.641651436 +0200
-@@ -126,3 +127,48 @@
- 0x800f746f KEY_PLAYER
- 0x800f7480 KEY_BRIGHTNESSDOWN
- 0x800f7481 KEY_PLAYPAUSE
-+
-+#Zotac AD10 Remote
-+0x8034048e KEY_POWER
-+0x8034043d KEY_POWER
-+0x80340400 KEY_NUMERIC_0
-+0x80340401 KEY_NUMERIC_1
-+0x80340402 KEY_NUMERIC_2
-+0x80340403 KEY_NUMERIC_3
-+0x80340404 KEY_NUMERIC_4
-+0x80340405 KEY_NUMERIC_5
-+0x80340406 KEY_NUMERIC_6
-+0x80340407 KEY_NUMERIC_7
-+0x80340408 KEY_NUMERIC_8
-+0x80340409 KEY_NUMERIC_9
-+0x80340432 KEY_NUMERIC_POUND
-+0x80340433 KEY_NUMERIC_STAR
-+0x8034043a KEY_DELETE
-+0x80340434 KEY_ENTER
-+0x8034045d KEY_MEDIA
-+0x8034040d KEY_MUTE
-+0x803404cb KEY_INFO
-+0x80340410 KEY_VOLUMEUP
-+0x80340411 KEY_VOLUMEDOWN
-+0x8034041e KEY_CHANNELUP
-+0x8034041f KEY_CHANNELDOWN
-+0x80340428 KEY_FASTFORWARD
-+0x80340429 KEY_REWIND
-+0x8034042c KEY_PLAY
-+0x80340437 KEY_RECORD
-+0x80340430 KEY_PAUSE
-+0x80340431 KEY_STOP
-+0x80340420 KEY_NEXT
-+0x80340421 KEY_PREVIOUS
-+0x80340458 KEY_UP
-+0x80340459 KEY_DOWN
-+0x8034045a KEY_LEFT
-+0x8034045b KEY_RIGHT
-+0x8034045c KEY_OK
-+0x80340483 KEY_EXIT
-+0x8034043e KEY_EPG
-+0x8034042f KEY_TITLE
-+0x8034046d KEY_RED
-+0x8034046e KEY_GREEN
-+0x8034046f KEY_YELLOW
-+0x80340470 KEY_BLUE
diff --git a/packages/sysutils/v4l-utils/patches/v4l-utils-20-add-samsung-keytable.patch b/packages/sysutils/v4l-utils/patches/v4l-utils-20-add-samsung-keytable.patch
new file mode 100644
index 0000000..bb64cc3
--- /dev/null
+++ b/packages/sysutils/v4l-utils/patches/v4l-utils-20-add-samsung-keytable.patch
@@ -0,0 +1,46 @@
+diff -Naur v4l-utils-1.8.1/utils/keytable/rc_keymaps/samsung v4l-utils-1.8.1.patch/utils/keytable/rc_keymaps/samsung
+--- v4l-utils-1.8.1/utils/keytable/rc_keymaps/samsung	1970-01-01 01:00:00.000000000 +0100
++++ v4l-utils-1.8.1.patch/utils/keytable/rc_keymaps/samsung	2015-11-18 12:21:30.202350149 +0100
+@@ -0,0 +1,31 @@
++# table samsung, type: NEC
++0x43532f KEY_NUMERIC_0
++0x435341 KEY_NUMERIC_1
++0x435342 KEY_NUMERIC_2
++0x435343 KEY_NUMERIC_3
++0x435344 KEY_NUMERIC_4
++0x435345 KEY_NUMERIC_5
++0x435346 KEY_NUMERIC_6
++0x435347 KEY_NUMERIC_7
++0x43533f KEY_NUMERIC_8
++0x435337 KEY_NUMERIC_9
++0x435305 KEY_ENTER
++0x435300 KEY_TITLE
++0x435326 KEY_INFO
++0x435323 KEY_VOLUMEUP
++0x43532b KEY_VOLUMEDOWN
++0x435375 KEY_FASTFORWARD
++0x435374 KEY_REWIND
++0x43537d KEY_PLAYPAUSE
++0x43530b KEY_STOP
++0x435313 KEY_NEXT
++0x43531b KEY_PREVIOUS
++0x43530d KEY_UP
++0x435315 KEY_DOWN
++0x435325 KEY_LEFT
++0x43531d KEY_RIGHT
++0x435336 KEY_MENU
++0x43531c KEY_EXIT
++0x43530f KEY_ZOOM
++0x43530e KEY_SUBTITLE
++0x43537e KEY_BACK
+diff -Naur v4l-utils-1.8.1/utils/keytable/rc_maps.cfg v4l-utils-1.8.1.patch/utils/keytable/rc_maps.cfg
+--- v4l-utils-1.8.1/utils/keytable/rc_maps.cfg	2015-09-19 20:54:17.000000000 +0200
++++ v4l-utils-1.8.1.patch/utils/keytable/rc_maps.cfg	2015-11-18 12:22:30.963487065 +0100
+@@ -112,6 +112,7 @@
+ *	rc-rc6-mce               rc6_mce
+ *	rc-real-audio-220-32-keys real_audio_220_32_keys
+ *	rc-reddo                 reddo
++*	rc-samsung               samsung
+ *	rc-snapstream-firefly    snapstream_firefly
+ *	rc-streamzap             streamzap
+ *	rc-su3000                su3000
diff --git a/packages/sysutils/v4l-utils/patches/v4l-utils-20-add_samsung_keytable.patch b/packages/sysutils/v4l-utils/patches/v4l-utils-20-add_samsung_keytable.patch
deleted file mode 100644
index bb64cc3..0000000
--- a/packages/sysutils/v4l-utils/patches/v4l-utils-20-add_samsung_keytable.patch
+++ /dev/null
@@ -1,46 +0,0 @@
-diff -Naur v4l-utils-1.8.1/utils/keytable/rc_keymaps/samsung v4l-utils-1.8.1.patch/utils/keytable/rc_keymaps/samsung
---- v4l-utils-1.8.1/utils/keytable/rc_keymaps/samsung	1970-01-01 01:00:00.000000000 +0100
-+++ v4l-utils-1.8.1.patch/utils/keytable/rc_keymaps/samsung	2015-11-18 12:21:30.202350149 +0100
-@@ -0,0 +1,31 @@
-+# table samsung, type: NEC
-+0x43532f KEY_NUMERIC_0
-+0x435341 KEY_NUMERIC_1
-+0x435342 KEY_NUMERIC_2
-+0x435343 KEY_NUMERIC_3
-+0x435344 KEY_NUMERIC_4
-+0x435345 KEY_NUMERIC_5
-+0x435346 KEY_NUMERIC_6
-+0x435347 KEY_NUMERIC_7
-+0x43533f KEY_NUMERIC_8
-+0x435337 KEY_NUMERIC_9
-+0x435305 KEY_ENTER
-+0x435300 KEY_TITLE
-+0x435326 KEY_INFO
-+0x435323 KEY_VOLUMEUP
-+0x43532b KEY_VOLUMEDOWN
-+0x435375 KEY_FASTFORWARD
-+0x435374 KEY_REWIND
-+0x43537d KEY_PLAYPAUSE
-+0x43530b KEY_STOP
-+0x435313 KEY_NEXT
-+0x43531b KEY_PREVIOUS
-+0x43530d KEY_UP
-+0x435315 KEY_DOWN
-+0x435325 KEY_LEFT
-+0x43531d KEY_RIGHT
-+0x435336 KEY_MENU
-+0x43531c KEY_EXIT
-+0x43530f KEY_ZOOM
-+0x43530e KEY_SUBTITLE
-+0x43537e KEY_BACK
-diff -Naur v4l-utils-1.8.1/utils/keytable/rc_maps.cfg v4l-utils-1.8.1.patch/utils/keytable/rc_maps.cfg
---- v4l-utils-1.8.1/utils/keytable/rc_maps.cfg	2015-09-19 20:54:17.000000000 +0200
-+++ v4l-utils-1.8.1.patch/utils/keytable/rc_maps.cfg	2015-11-18 12:22:30.963487065 +0100
-@@ -112,6 +112,7 @@
- *	rc-rc6-mce               rc6_mce
- *	rc-real-audio-220-32-keys real_audio_220_32_keys
- *	rc-reddo                 reddo
-+*	rc-samsung               samsung
- *	rc-snapstream-firefly    snapstream_firefly
- *	rc-streamzap             streamzap
- *	rc-su3000                su3000
diff --git a/packages/sysutils/v4l-utils/patches/v4l-utils-21-add-newer-hp-mce-remote.patch b/packages/sysutils/v4l-utils/patches/v4l-utils-21-add-newer-hp-mce-remote.patch
index 8de1498..c9050e1 100644
--- a/packages/sysutils/v4l-utils/patches/v4l-utils-21-add-newer-hp-mce-remote.patch
+++ b/packages/sysutils/v4l-utils/patches/v4l-utils-21-add-newer-hp-mce-remote.patch
@@ -1,13 +1,10 @@
-diff --git a/utils/keytable/rc_keymaps/rc6_mce b/utils/keytable/rc_keymaps/rc6_mce
-index d469845..c833a0c 100644
---- a/utils/keytable/rc_keymaps/rc6_mce
-+++ b/utils/keytable/rc_keymaps/rc6_mce
-@@ -173,3 +173,69 @@
- 0x8034046e KEY_GREEN
- 0x8034046f KEY_YELLOW
- 0x80340470 KEY_BLUE
-+
-+#HP MCE Remote
+diff --git a/utils/keytable/rc_keymaps/hp_mce b/utils/keytable/rc_keymaps/hp_mce
+new file mode 100644
+index 0000000..266dc8a
+--- /dev/null
++++ b/utils/keytable/rc_keymaps/hp_mce
+@@ -0,0 +1,65 @@
++# table hp_mce, type: RC6
 +0x80111200 KEY_NUMERIC_0
 +0x80111201 KEY_NUMERIC_1
 +0x80111202 KEY_NUMERIC_2
diff --git a/packages/sysutils/v4l-utils/patches/v4l-utils-22-add-XBOX_One-remote.patch b/packages/sysutils/v4l-utils/patches/v4l-utils-22-add-XBOX_One-remote.patch
deleted file mode 100644
index db4f120..0000000
--- a/packages/sysutils/v4l-utils/patches/v4l-utils-22-add-XBOX_One-remote.patch
+++ /dev/null
@@ -1,36 +0,0 @@
-diff -Naur v4l-utils-0.8.9/utils/keytable/rc_keymaps/rc6_mce v4l-utils-0.8.9.patch/utils/keytable/rc_keymaps/rc6_mce
---- v4l-utils-0.8.9/utils/keytable/rc_keymaps/rc6_mce	2014-05-20 14:31:04.585559227 +0200
-+++ v4l-utils-0.8.9.patch/utils/keytable/rc_keymaps/rc6_mce	2014-05-20 17:47:04.341378915 +0200
-@@ -1,4 +1,4 @@
--# table rc6_mce, type: RC6
-+# table rc6_mce, type: RC6 NEC
- 0x800f0400 KEY_NUMERIC_0
- 0x800f0401 KEY_NUMERIC_1
- 0x800f0402 KEY_NUMERIC_2
-@@ -239,3 +239,26 @@
- 0x8011129f KEY_LASTSOURCE
- 0x8011128b KEY_SOURCE
- 0x801112a3 KEY_SCREEN
-+
-+#XBOX One Remote (Protocol: NEC)
-+0x80d864 KEY_MEDIA
-+0x80d820 KEY_LEFT
-+0x80d821 KEY_RIGHT
-+0x80d81e KEY_UP
-+0x80d81f KEY_DOWN
-+0x80d822 KEY_OK
-+0x80d823 KEY_EXIT
-+0x80d80e KEY_MUTE
-+0x80d810 KEY_VOLUMEUP
-+0x80d811 KEY_VOLUMEDOWN
-+0x80d812 KEY_CHANNELUP
-+0x80d813 KEY_CHANNELDOWN
-+0x80d814 KEY_FASTFORWARD
-+0x80d815 KEY_REWIND
-+0x80d870 KEY_PLAYPAUSE
-+0x80d819 KEY_STOP
-+0x80d81a KEY_NEXT
-+0x80d81b KEY_PREVIOUS
-+0x80d826 KEY_INFO
-+0x80d86f KEY_EPG
-+0x80d86e KEY_ZOOM
diff --git a/packages/sysutils/v4l-utils/patches/v4l-utils-22-add-xbox-one-remote.patch b/packages/sysutils/v4l-utils/patches/v4l-utils-22-add-xbox-one-remote.patch
new file mode 100644
index 0000000..1b9c808
--- /dev/null
+++ b/packages/sysutils/v4l-utils/patches/v4l-utils-22-add-xbox-one-remote.patch
@@ -0,0 +1,28 @@
+diff --git a/utils/keytable/rc_keymaps/xbox_one b/utils/keytable/rc_keymaps/xbox_one
+new file mode 100644
+index 0000000..07614ee
+--- /dev/null
++++ b/utils/keytable/rc_keymaps/xbox_one
+@@ -0,0 +1,22 @@
++# table xbox_one, type: NEC
++0x80d864 KEY_MEDIA
++0x80d820 KEY_LEFT
++0x80d821 KEY_RIGHT
++0x80d81e KEY_UP
++0x80d81f KEY_DOWN
++0x80d822 KEY_OK
++0x80d823 KEY_EXIT
++0x80d80e KEY_MUTE
++0x80d810 KEY_VOLUMEUP
++0x80d811 KEY_VOLUMEDOWN
++0x80d812 KEY_CHANNELUP
++0x80d813 KEY_CHANNELDOWN
++0x80d814 KEY_FASTFORWARD
++0x80d815 KEY_REWIND
++0x80d870 KEY_PLAYPAUSE
++0x80d819 KEY_STOP
++0x80d81a KEY_NEXT
++0x80d81b KEY_PREVIOUS
++0x80d826 KEY_INFO
++0x80d86f KEY_EPG
++0x80d86e KEY_ZOOM
diff --git a/packages/sysutils/v4l-utils/patches/v4l-utils-23-add-cubox-i-remote.patch b/packages/sysutils/v4l-utils/patches/v4l-utils-23-add-cubox-i-remote.patch
index abf9aaa..1f54a6e 100644
--- a/packages/sysutils/v4l-utils/patches/v4l-utils-23-add-cubox-i-remote.patch
+++ b/packages/sysutils/v4l-utils/patches/v4l-utils-23-add-cubox-i-remote.patch
@@ -1,12 +1,10 @@
-diff -Nru a/utils/keytable/rc_keymaps/rc6_mce b/utils/keytable/rc_keymaps/rc6_mce
---- a/utils/keytable/rc_keymaps/rc6_mce	2015-02-17 12:25:15.452561904 +0200
-+++ b/utils/keytable/rc_keymaps/rc6_mce	2015-05-27 13:22:28.089766130 +0300
-@@ -262,3 +262,15 @@
- 0x80d826 KEY_INFO
- 0x80d86f KEY_EPG
- 0x80d86e KEY_ZOOM
-+
-+#CuBox-i and HummingBoard remote control
+diff --git a/utils/keytable/rc_keymaps/cuboxi b/utils/keytable/rc_keymaps/cubox_i
+new file mode 100644
+index 0000000..1063253
+--- /dev/null
++++ b/utils/keytable/rc_keymaps/cubox_i
+@@ -0,0 +1,10 @@
++# table cubox_i, type: RC6
 +0x1f020b0 KEY_MEDIA
 +0x1f02078 KEY_UP
 +0x1f02044 KEY_ENTER
@@ -16,4 +14,3 @@ diff -Nru a/utils/keytable/rc_keymaps/rc6_mce b/utils/keytable/rc_keymaps/rc6_mc
 +0x1f020c4 KEY_BACKSPACE
 +0x1f02068 KEY_PLAYPAUSE
 +0x1f02064 KEY_EPG
-+
diff --git a/packages/sysutils/v4l-utils/patches/v4l-utils-24-add-xbox-360-remote.patch b/packages/sysutils/v4l-utils/patches/v4l-utils-24-add-xbox-360-remote.patch
new file mode 100644
index 0000000..85119d5
--- /dev/null
+++ b/packages/sysutils/v4l-utils/patches/v4l-utils-24-add-xbox-360-remote.patch
@@ -0,0 +1,72 @@
+diff --git a/utils/keytable/rc_keymaps/xbox_360 b/utils/keytable/rc_keymaps/xbox_360
+new file mode 100644
+index 0000000..b90ecfd
+--- /dev/null
++++ b/utils/keytable/rc_keymaps/xbox_360
+@@ -0,0 +1,66 @@
++# table xbox_360, type: RC6
++0x800f7400 KEY_NUMERIC_0
++0x800f7401 KEY_NUMERIC_1
++0x800f7402 KEY_NUMERIC_2
++0x800f7403 KEY_NUMERIC_3
++0x800f7404 KEY_NUMERIC_4
++0x800f7405 KEY_NUMERIC_5
++0x800f7406 KEY_NUMERIC_6
++0x800f7407 KEY_NUMERIC_7
++0x800f7408 KEY_NUMERIC_8
++0x800f7409 KEY_NUMERIC_9
++0x800f740a KEY_DELETE
++0x800f740b KEY_ENTER
++0x800f740c KEY_SLEEP
++0x800f740d KEY_MEDIA
++0x800f7464 KEY_MEDIA
++0x800f740e KEY_MUTE
++0x800f740f KEY_INFO
++0x800f7410 KEY_VOLUMEUP
++0x800f7411 KEY_VOLUMEDOWN
++0x800f7412 KEY_CHANNELUP
++0x800f746C KEY_CHANNELUP
++0x800f7413 KEY_CHANNELDOWN
++0x800f746D KEY_CHANNELDOWN
++0x800f7414 KEY_FASTFORWARD
++0x800f7415 KEY_REWIND
++0x800f7416 KEY_PLAY
++0x800f7417 KEY_RECORD
++0x800f7418 KEY_PAUSE
++0x800f7419 KEY_STOP
++0x800f741a KEY_NEXT
++0x800f741b KEY_PREVIOUS
++0x800f741c KEY_NUMERIC_POUND
++0x800f741d KEY_NUMERIC_STAR
++0x800f741e KEY_UP
++0x800f741f KEY_DOWN
++0x800f7420 KEY_LEFT
++0x800f7421 KEY_RIGHT
++0x800f7422 KEY_OK
++0x800f7423 KEY_EXIT
++0x800f7424 KEY_DVD
++0x800f744f KEY_EPG
++0x800f7427 KEY_ZOOM
++0x800f7432 KEY_MODE
++0x800f7433 KEY_PRESENTATION
++0x800f7428 KEY_EJECTCD
++0x800f743a KEY_BRIGHTNESSUP
++0x800f7446 KEY_TV
++0x800f7447 KEY_AUDIO
++0x800f7448 KEY_PVR
++0x800f7449 KEY_CAMERA
++0x800f744a KEY_VIDEO
++0x800f744c KEY_LANGUAGE
++0x800f7451 KEY_TITLE
++0x800f744e KEY_PRINT
++0x800f7450 KEY_RADIO
++0x800f745a KEY_SUBTITLE
++0x800f7425 KEY_RED
++0x800f7466 KEY_GREEN
++0x800f7426 KEY_YELLOW
++0x800f7468 KEY_BLUE
++0x800f7465 KEY_POWER2
++0x800f746e KEY_PLAYPAUSE
++0x800f746f KEY_PLAYER
++0x800f7480 KEY_BRIGHTNESSDOWN
++0x800f7481 KEY_PLAYPAUSE
diff --git a/packages/sysutils/v4l-utils/patches/v4l-utils-25-add-zotac-ad10-remote.patch b/packages/sysutils/v4l-utils/patches/v4l-utils-25-add-zotac-ad10-remote.patch
new file mode 100644
index 0000000..5e6af84
--- /dev/null
+++ b/packages/sysutils/v4l-utils/patches/v4l-utils-25-add-zotac-ad10-remote.patch
@@ -0,0 +1,50 @@
+diff --git a/utils/keytable/rc_keymaps/zotac_ad10 b/utils/keytable/rc_keymaps/zotac_ad10
+new file mode 100644
+index 0000000..e93fad4
+--- /dev/null
++++ b/utils/keytable/rc_keymaps/zotac_ad10
+@@ -0,0 +1,44 @@
++# table zotac_ad10, type: RC6
++0x8034048e KEY_POWER
++0x8034043d KEY_POWER
++0x80340400 KEY_NUMERIC_0
++0x80340401 KEY_NUMERIC_1
++0x80340402 KEY_NUMERIC_2
++0x80340403 KEY_NUMERIC_3
++0x80340404 KEY_NUMERIC_4
++0x80340405 KEY_NUMERIC_5
++0x80340406 KEY_NUMERIC_6
++0x80340407 KEY_NUMERIC_7
++0x80340408 KEY_NUMERIC_8
++0x80340409 KEY_NUMERIC_9
++0x80340432 KEY_NUMERIC_POUND
++0x80340433 KEY_NUMERIC_STAR
++0x8034043a KEY_DELETE
++0x80340434 KEY_ENTER
++0x8034045d KEY_MEDIA
++0x8034040d KEY_MUTE
++0x803404cb KEY_INFO
++0x80340410 KEY_VOLUMEUP
++0x80340411 KEY_VOLUMEDOWN
++0x8034041e KEY_CHANNELUP
++0x8034041f KEY_CHANNELDOWN
++0x80340428 KEY_FASTFORWARD
++0x80340429 KEY_REWIND
++0x8034042c KEY_PLAY
++0x80340437 KEY_RECORD
++0x80340430 KEY_PAUSE
++0x80340431 KEY_STOP
++0x80340420 KEY_NEXT
++0x80340421 KEY_PREVIOUS
++0x80340458 KEY_UP
++0x80340459 KEY_DOWN
++0x8034045a KEY_LEFT
++0x8034045b KEY_RIGHT
++0x8034045c KEY_OK
++0x80340483 KEY_EXIT
++0x8034043e KEY_EPG
++0x8034042f KEY_TITLE
++0x8034046d KEY_RED
++0x8034046e KEY_GREEN
++0x8034046f KEY_YELLOW
++0x80340470 KEY_BLUE
diff --git a/packages/sysutils/v4l-utils/patches/v4l-utils-50-use-multi-table-add-fake-tables.patch b/packages/sysutils/v4l-utils/patches/v4l-utils-50-use-multi-table-add-fake-tables.patch
new file mode 100644
index 0000000..4e63d00
--- /dev/null
+++ b/packages/sysutils/v4l-utils/patches/v4l-utils-50-use-multi-table-add-fake-tables.patch
@@ -0,0 +1,29 @@
+diff --git a/utils/keytable/rc_maps.cfg b/utils/keytable/rc_maps.cfg
+index e69fd6a..c0fe122 100644
+--- a/utils/keytable/rc_maps.cfg
++++ b/utils/keytable/rc_maps.cfg
+@@ -109,7 +109,9 @@
+ *	rc-proteus-2309          proteus_2309
+ *	rc-purpletv              purpletv
+ *	rc-pv951                 pv951
+-*	rc-rc6-mce               rc6_mce
++# use combined multi-table on MCE receivers
++#*	rc-rc6-mce               rc6_mce
++*	rc-rc6-mce               libreelec_multi
+ *	rc-real-audio-220-32-keys real_audio_220_32_keys
+ *	rc-reddo                 reddo
+ *	rc-snapstream-firefly    snapstream_firefly
+@@ -136,6 +138,13 @@
+ *	rc-videomate-tv-pvr      videomate_tv_pvr
+ *	rc-winfast-usbii-deluxe  winfast_usbii_deluxe
+ *	rc-winfast               winfast
++# fake tables for LibreELEC keymaps
++*	rc-cubox-i               cubox-i
++*	rc-hp-mce                hp_mce
++*	rc-samsung               samsung
++*	rc-xbox-360              xbox_360
++*	rc-xbox-one              xbox_one
++*	rc-zotac-ad10            zotac_ad10
+ # *	*			 a800                 # found in a800.c
+ # *	*			 af9005               # found in af9005-remote.c
+ # *	*			 cinergyt2            # found in cinergyT2-core.c

From a88314530c32048bc9705f4fd738f2f8dbab7943 Mon Sep 17 00:00:00 2001
From: Matthias Reichl <hias@horus.com>
Date: Sun, 12 Mar 2017 17:52:08 +0100
Subject: [PATCH 2/2] v4l-utils: fix invalid keynames in hp mce remote table

Signed-off-by: Matthias Reichl <hias@horus.com>
---
 .../v4l-utils/patches/v4l-utils-21-add-newer-hp-mce-remote.patch    | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/packages/sysutils/v4l-utils/patches/v4l-utils-21-add-newer-hp-mce-remote.patch b/packages/sysutils/v4l-utils/patches/v4l-utils-21-add-newer-hp-mce-remote.patch
index c9050e1..08cc28d 100644
--- a/packages/sysutils/v4l-utils/patches/v4l-utils-21-add-newer-hp-mce-remote.patch
+++ b/packages/sysutils/v4l-utils/patches/v4l-utils-21-add-newer-hp-mce-remote.patch
@@ -65,7 +65,7 @@ index 0000000..266dc8a
 +0x7011126e KEY_PLAYPAUSE
 +0x8011120b KEY_MEDIA
 +0x70111280 KEY_BRIGHTNESSDOWN
-+0x80111254 KEY_SETTINGS
-+0x8011129f KEY_LASTSOURCE
-+0x8011128b KEY_SOURCE
++0x80111254 KEY_MENU
++0x8011129f KEY_LAST
++0x8011128b KEY_SETUP
 +0x801112a3 KEY_SCREEN
